grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -92,6 +87,21 @@ class UserService {
         return userInstance.save(failOnError: true)
     }
 
+    void changePassword(User user, String password, String passwordConfirm) {
+        if (!canUserEditPassword(AuthService.currentUser, user)) {
+            String errorMessage = applicationTagLib.message(
+                    code: 'errors.accessDenied.message',
+                    default: 'Apologies, but you are not authorized to access this page or to perform this action.',
+            )
+            throw new AuthenticationException(errorMessage)
+        }
+
+        if (user.password != password) {
+            user.password = password?.encodeAsPassword()
+            user.passwordConfirm = passwordConfirm?.encodeAsPassword()
+            user.save(failOnError: true)
```

Comment:
an alternative way could be to use the save from the data service, as it throws validation exception when validation fails by default. (no need to change, just titbit).